src/EnergyPlus/api/runtime.cc

```diff
@@ -83,147 +90,180 @@ void issueText(const char * message) {
     EnergyPlus::ShowContinueError(message);
 }
 
-void registerProgressCallback(void (*f)(int const)) {
+void registerProgressCallback(EnergyPlusState EP_UNUSED(state), void (*f)(int const)) {
+    // auto *this_state = reinterpret_cast<EnergyPlus::EnergyPlusData *>(state);
     EnergyPlus::DataGlobals::progressCallback = f;
```

Comment:
I did *not* convert over the older progressCallback and messageCallback functions to accept a state object argument.  This would have meant changing DisplayString and other Display* functions to accept a state object argument.  Which means like every single function in the entire codebase would've needed state.  I need to decide the best path here.